weather-app/src/app/landing-page/landing-page.component.ts

```diff
@@ -1,22 +1,42 @@
 import { Component, OnInit } from '@angular/core';
+import { byCountry } from 'country-code-lookup';
+import { WeatherService } from '../services/weather.service';
 
 @Component({
   selector: 'app-landing-page',
   templateUrl: './landing-page.component.html',
   styleUrls: ['./landing-page.component.scss', '../../styles/_container.scss']
 })
 export class LandingPageComponent implements OnInit {
+  public currentCity: string = '';
+  public currentCountry: string = '';
+  public currentDay: string = '';
+  public currentTime: string = ''; 
+  public currentDegrees: string = ''; 
+  public curentWeatherIconUrl = '';
+ 
   isDarkMode = false;
-  currentCity = 'Ivano-Frankivsk';
-  currentCountry = 'UA';
-  currentDay = 'Monday';
-  currentTime = '06:00';
-  currentDegrees = '+12';
-  curentWeatherIconUrl = 'https://raw.githubusercontent.com/basmilius/weather-icons/029d142b34871bcbc90d7cd46081c50310c831c5/production/fill/svg/overcast-day-fog.svg';
-  
-  constructor() {
+  weekday = ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"];
+
+  constructor(private weatherService: WeatherService) {
   }
 
   ngOnInit(): void {
+    this.weatherService.getCurrentWeather('Lviv').subscribe((data: any) => {
+      const currentDate = data.location.localtime.split(" ")[0];
+      const currentDayIndex = new Date(currentDate).getDay();
+      this.currentCity = data.location.name;
+      this.currentCountry = byCountry(data.location.country)?.iso2 ?? 'null';
+      this.currentDay = this.weekday[currentDayIndex + 1];
+      this.currentTime = data.location.localtime.split(" ")[1];
+      this.currentDegrees = data.current.temp_c > 0 ? '+' + data.current.temp_c : data.current.temp_c;
```

Comment:
I'd replace `'+' + data.current.temp_c` with ``+${data.current.temp_c}`` to make it more readable.
'+' + looks awkward right now.

and I also assume that you would want to have `: -${data.current.temp_c}` for the temperature below 0 degrees.

Reply:
Fixed.